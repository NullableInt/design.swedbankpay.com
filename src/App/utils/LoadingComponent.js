import React, { Component } from "react";
import Loader from "@/Loader";
import { loader } from "$/px-script";

class LoadingComponent extends Component {
    componentDidMount () {
        loader.init();
    }

    render () {
        return (
            <div id="loading-component">
                <Loader size="large" muted />
                <span className="loading-text text-muted">Loading...</span>
            </div>
        );
    }
}

export default LoadingComponent;
