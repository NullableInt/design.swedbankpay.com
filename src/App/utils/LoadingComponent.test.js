import React from "react";
import { shallow } from "enzyme";

import LoadingComponent from "./LoadingComponent";

describe("App utils: LoadingComponent", () => {
    it("is defined", () => {
        expect(LoadingComponent).toBeDefined();
    });

    it("renders", () => {
        console.warn = jest.fn();

        const wrapper = shallow(<LoadingComponent />);

        expect(wrapper).toMatchSnapshot();
        expect(console.warn).toHaveBeenCalled();
    });
});
