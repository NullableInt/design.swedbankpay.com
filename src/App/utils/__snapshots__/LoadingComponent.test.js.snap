// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App utils: LoadingComponent renders 1`] = `
<div
  id="loading-component"
>
  <Loader
    muted={true}
    size="large"
  />
  <span
    className="loading-text text-muted"
  >
    Loading...
  </span>
</div>
`;
